postmortems/2024-04-01-tx-receipts-validation-skip-incident.md

```diff
@@ -0,0 +1,61 @@
+# 2024-04-03 Chain Halt due to tx receipts validation skip
+
+# Incident Summary
+
+On April 1, 2024, at 02:51:15 UTC, a wrong block was generated at block number [9029744](https://kromascan.com/block/9029744).
+
+Some nodes ignored the batch due to a hash discrepancy in that block compared to the block hash of the sequencer. This 
+discrepancy was caused due to that the sequencer omitted a deposited transaction from L1.
+
+The sequencer is recovered by rollback to block number 9029744. Block generation was halted for about 3 hours.
+
+# Background
+
+We operate our own L1 RPC nodes and have duplicated L1 RPC to ensure stable operation of L2 sequencer and RPC nodes. If 
+the block head of the active L1 RPC falls behind a certain threshold, it is configured to look towards the fallback RPC 
+node. Additionally, all kroma-nodes, including sequencers, have been running with the --l1.trustrpc option, using our L1
+RPC as the trust RPC through this option.
+
+# Causes
+
+Block head of our L1 RPC began to fall behind the canonical chain, triggering a switch to the fallback L1 RPC once it 
+deviated beyond a certain threshold. During the switch, sequencer attempted to retrieve transaction receipts from the L1
+RPC node that fell behind it had not yet synced, resulting in empty transaction receipt responses.
+
+Actually, the block in question did contain [deposited transaction](https://kromascan.com/tx/0xf76e4f34645bc3e172909fb03311cd1770f9c574ce46bd1da3644f9cea82c0e3)
+on L1, yet the sequencer proceeded to generate the block without processing this transaction, having received empty 
+transaction receipts from the trusted L1 RPC. As a consequence, starting from block 9029744, a different block was 
+generated, leading to a generation of wrong block. This is the transaction not included in the block sequencer 
```

Comment:
I think a link of the omitted transaction to `This` is missing.

Reply:
Added in [47009b1](https://github.com/kroma-network/kroma/pull/289/commits/47009b1023eb7a05e741326b5c955428aa1ebe88)